github/orgs_personal_access_tokens.go

```diff
@@ -11,6 +11,69 @@ import (
 	"net/http"
 )
 
+// PersonalAccessToken represents the minimal representation of an organization programmatic access grant.
+//
+// GitHub API docs: https://docs.github.com/en/rest/orgs/personal-access-tokens?apiVersion=2022-11-28
+type PersonalAccessToken struct {
+	// "Unique identifier of the fine-grained personal access token.
+	// The `pat_id` used to get details about an approved fine-grained personal access token.
+	ID *int64 `json:"id"`
+
+	// Owner is the GitHub user associated with the token.
+	Owner *User `json:"owner"`
+
+	// RepositorySelection is the type of repository selection requested.
+	// Possible values are: "none", "all", "subset".
+	RepositorySelection *string `json:"repository_selection"`
+
+	// URL to the list of repositories the fine-grained personal access token can access.
+	// Only follow when `repository_selection` is `subset`.
+	RepositoriesURL *string `json:"repositories_url"`
+
+	// Permissions are the permissions requested, categorized by type.
+	Permissions *PersonalAccessTokenPermissions `json:"permissions"`
+
+	// Date and time when the fine-grained personal access token was approved to access the organization.
+	AccessGrantedAt *Timestamp `json:"access_granted_at"`
+
+	// Whether the associated fine-grained personal access token has expired.
+	TokenExpired *bool `json:"token_expired"`
+
+	// Date and time when the associated fine-grained personal access token expires.
+	TokenExpiresAt *Timestamp `json:"token_expires_at"`
+
+	// Date and time when the associated fine-grained personal access token was last used for authentication.
+	TokenLastUsedAt *Timestamp `json:"token_last_used_at"`
+}
+
+// ListFineGrainedPersonalAccessTokens lists approved fine-grained personal access tokens owned by organization members that can access organization resources.
+// Only GitHub Apps can call this API, using the `Personal access tokens` organization permissions (read).
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/personal-access-tokens#list-fine-grained-personal-access-tokens-with-access-to-organization-resources
+//
+//meta:operation GET /orgs/{org}/personal-access-tokens
+func (s *OrganizationsService) ListFineGrainedPersonalAccessTokens(ctx context.Context, org string, opts *ListOptions) ([]*PersonalAccessToken, *Response, error) {
```

Comment:
This endpoint can take a whole bunch more options:

```
sort stringThe property by which to sort the results.Default: created_atValue: created_at
--
direction stringThe direction to sort the results by.Default: descCan be one of: asc, desc
owner arrayA list of owner usernames to use to filter the results.
repository stringThe name of the repository to use to filter the results.
permission stringThe permission to use to filter the results.
last_used_before stringOnly show fine-grained personal access tokens used before the given time. This is a timestamp in ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ.
last_used_after stringOnly show fine-grained personal access tokens used after the given time. This is a timestamp in ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ.

sort string
The property by which to sort the results.

Default: created_at

Value: created_at

direction string
The direction to sort the results by.

Default: desc

Can be one of: asc, desc

owner array
A list of owner usernames to use to filter the results.

repository string
The name of the repository to use to filter the results.

permission string
The permission to use to filter the results.

last_used_before string
Only show fine-grained personal access tokens used before the given time. This is a timestamp in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format: YYYY-MM-DDTHH:MM:SSZ.

last_used_after string
Only show fine-grained personal access tokens used after the given time. This is a timestamp in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format: YYYY-MM-DDTHH:MM:SSZ.
```

So let's please create a new `type ListFineGrainedPATOptions struct { ... ListOptions }` that can take all these parameter, and make sure to add `omitempty` to each one of them since they are optional.

Reply:
Good point and sorry for missing that. I added that on my latest commit.

One thing though, using the `owner` array as 

```go
Owner           []string `url:"owner,omitempty,comma"`
```

will cause the API to return 422 status code, as it expects the `[]owner=...` format, so I had to create the helper function below.

I tried to find an example on the codebase and I found `IssueListOptions`, but this one seems to work as that endpoint expects the `orgs/{ORG}/issues?labels=bug%2Cfeature` format

Reply:
@arymoraes - I'm looking all over the GitHub v3 API documentation and can't find `[]owner=...` anywhere.
Where did you find this?  It seems incredibly strange syntax to me for a query parameter.

Reply:
I also found it very strange. I did not find it on the docs, but by trial and error (and by remembering seeing this kind of issue once). 
If you run a CURL with:

curl -L \
```
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: {TOKEN}" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/orgs/{ORG}/personal-access-tokens?owner={username}
```

It will give you a `422 status code error`:

```json
{
  "message": "Invalid request.\n\nInvalid input: `\"{username}\"` is not of type `array`.",
  "documentation_url": "https://docs.github.com/rest/orgs/personal-access-tokens#list-fine-grained-personal-access-tokens-with-access-to-organization-resources",
  "status": "422"
}
```

However:

```
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer {TOKEN}" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/orgs/{ORG}/personal-access-tokens?owner[]={username}
```

Is valid and will indeed return you a list of tokens filtered by that owner

Reply:
OK, wow, that is bizarre... so I was hoping that your unit test would show how multiple owners would appear in the resulting URL, but it doesn't... do they show up as comma-separated values?

Reply:
Honestly, it gets even weirder if you want multiple owners.

```
personal-access-tokens?owner[]=owner1,owner2
```

doesn't work, it'll just return an empty array of tokens (maybe if thinks the owner name is `owner1,owner2`?

You have to do

```
personal-access-tokens?owner[]=owner1&owner[]=owner2
```

To be able to filter tokens from any of those two owners. This is the one that is generated with `addListFineGrainedPATOptions`.

I am finding this really weird, but this is the way it works even with curl, so I'm assuming this is indeed the way that GitHub API wants us to send the query param. I tried to search for others with the same question, but could not. I could take another look though.

I should probably improve the comment for `addListFineGrainedPATOptions` to add the multiple owners scenario, and maybe also test the URL explicitly on the unit test?

Reply:
Wow, that's amazing... thank you for the details, @arymoraes !

Maybe could you please just add a comment explaining all this weirdness would be extremely helpful for anyone else that comes across this and tries to use it.  Thank you!

Reply:
Done!

I added the comments to the `addListFineGrainedPATOptions` helper function declaration, as well as inside `addListFineGrainedPATOptions` when we call `addListFineGrainedPATOptions`, so whoever runs into that code knows why we are doing this.